scripts/pylib/twister/twisterlib/harness.py

```diff
@@ -507,6 +487,116 @@ def _parse_report_file(self, report):
             self.instance.reason = 'No tests collected'
 
 
+class Robotframework(TwisterLibraryHarness):
+
+    def configure(self, instance: TestInstance):
+        super(Robotframework, self)._configure(instance, "robot", 'robot_report.xml', 'robot.html')
+
+    def robot_run(self, timeout):
+        self._run(timeout)
+
+    def _get_base_command(self):
+        command = [
+            sys.executable,
+            "-m",
+            "twister_harness.robot_framework.robot_wrapper",
+            "--log",
+            self.log_file_path,
+            "--xunit",
+            self.report_file,
+            f'--outputdir={self.running_dir}/robot',
+             f'--build-dir={self.running_dir}',
+        ]
+        return command
+
+    def _add_toolspecific_flags(self, handler: Handler, command: list[str]) -> None:
+        config = self.instance.testsuite.harness_config
+        robot_test_path = config.get('robot_test_path', 'robot') if config else 'robot'
+        robot_args_yaml = config.get('robot_args', []) if config else []
+
+        if handler.options.robot_args:
+            for extra_args in handler.options.robot_args:
+                command.extend(shlex.split(extra_args))
+
+            if robot_args_yaml:
+                logger.warning(f'The robot_args ({handler.options.robot_args}) specified '
+                               'in the command line will override the robot_args defined '
+                               f'in the YAML file {robot_args_yaml}')
+        else:
+            command.extend(robot_args_yaml)
+
+        if handler.options.verbose > 1:
+            command.extend([
+                '--loglevel', 'DEBUG',
+            ])
+
+        if not PYTEST_PLUGIN_INSTALLED:
+            pytest_plugin_path = os.path.join(ZEPHYR_BASE, 'scripts', 'pylib',
+                                              'pytest-twister-harness',
+                                              'src')
+
+            command.extend(['--pythonpath', pytest_plugin_path])
```

Comment:
probably do not need it after calling `_update_command_with_env_dependencies` (is called in `run_command` and adds twister-harness dir to PYTHONPATHS)